xen/arch/arm/coproc/plat/coproc_xxx.c

```diff
@@ -122,51 +144,30 @@ static const struct coproc_ops vcoproc_xxx_vcoproc_ops = {
     .ctx_switch_to       = vcoproc_xxx_ctx_switch_to,
 };
 
-static void coproc_xxx_irq_handler(int irq, void *dev,
-                                   struct cpu_user_regs *regs)
-{
-    struct coproc_device *coproc_xxx = dev;
-
-    (void)coproc_xxx;
-}
-
 static int coproc_xxx_dt_probe(struct dt_device_node *np)
 {
-    struct coproc_device *coproc_xxx;
+    struct mcoproc_device *coproc_xxx;
     struct device *dev = &np->dev;
-    int i, ret;
+    int ret;
 
-    coproc_xxx = coproc_alloc(np, &vcoproc_xxx_vcoproc_ops);
+    coproc_xxx = coproc_alloc(np,  &coproc_xxx_desc, &vcoproc_xxx_vcoproc_ops);
     if ( IS_ERR_OR_NULL(coproc_xxx) )
-        return PTR_ERR(coproc_xxx);
-
-    for ( i = 0; i < coproc_xxx->num_irqs; ++i )
     {
-        ret = request_irq(coproc_xxx->irqs[i],
-                         IRQF_SHARED,
-                         coproc_xxx_irq_handler,
-                         "coproc_xxx irq",
-                         coproc_xxx);
-        if ( ret )
-        {
-            COPROC_ERROR(dev, "failed to request irq %d (%u)\n", i,
-                         coproc_xxx->irqs[i]);
-            goto out_release_irqs;
-        }
+        COPROC_DEBUG(dev, "failed to allocate coproc\n");
+        ret = PTR_ERR(coproc_xxx);
+        goto out;
```

Comment:
Why goto out here, but not just return error? Do we need release coproc if we couldn't allocate it?